File: qc-domain/src/main/java/com/eager/questioncloud/domain/verification/Email.java
package com.eager.questioncloud.domain.verification;

import lombok.Getter;

@Getter
public class Email {
    private String to;
    private String subject;
    private String content;

    public Email(String to, String subject, String content) {
        this.to = to;
        this.subject = subject;
        this.content = content;
    }

    public static Email of(EmailVerification emailVerification) {
        EmailVerificationTemplate template = EmailVerificationTemplateCreator.getTemplate(
            emailVerification.getEmailVerificationType(),
            emailVerification.getToken());

        return new Email(emailVerification.getEmail(), template.getTitle(), template.getContent());
    }
}

File: qc-core/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/domain/Email.kt
package com.eager.questioncloud.user.domain

import com.eager.questioncloud.user.template.EmailVerificationTemplateCreator

class Email(
    val to: String,
    val subject: String,
    val content: String
) {
    companion object {
        fun of(emailVerification: EmailVerification): Email {
            val template = EmailVerificationTemplateCreator.getTemplate(
                emailVerification.emailVerificationType,
                emailVerification.token
            )

            return Email(
                emailVerification.email,
                template.title,
                template.content
            )
        }
    }
}
